omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java

```diff
@@ -319,11 +320,14 @@ private boolean testOperationImplemented(OperationEnum operation, DelegatingReso
 					} else {
 						try {
 							// to avoid saving data to the database, we pass a null SimpleObject
-							method.invoke(resourceHandler, null, RestConstants.SWAGGER_IMPOSSIBLE_UNIQUE_ID,
+							method.invoke(resourceHandler, RestConstants.SWAGGER_IMPOSSIBLE_UNIQUE_ID, null,
 							    new RequestContext());
 						}
-						catch (ResourceDoesNotSupportOperationException re) {
```

Comment:
How about for cases which throw ResourceDoesNotSupportOperationException?

Reply:
Actually it's never thrown here. It gets wrapped in the InvocationTargetException in the underlying invoked method.

https://stackoverflow.com/questions/6020719/what-could-cause-java-lang-reflect-invocationtargetexception

Reply:
How sure are you that it never gets thrown? Looking at the fact the the original author probably had it thrown to put this catch clause. Most probably in some cases.

Reply:
I'm so sure with this. 
When the `ResourceDoesNotSupportOperationException` is thrown in the invoked method eg. `#create(args..)` that causes the `InvocationTargetException`. 
So if we want the original Exception(cause), we invoke `e.getCause()`.

Please take time and look at the docs : https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/InvocationTargetException.html